### PR TITLE
Documentation Changes: README and Site Docs

### PR DESCRIPTION
**Pull request by diesalbla**
We extend the microsite documentation. 

- We add a microsite page of references, that is to say, conference talks, blog posts, or academic papers, related to FS2.
- Moves the section of libraries using FS2 to another _ecosystem_ page in the microsite. 
- We shorten those sections from the README alredy covered by these pages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
 [![Latest version](https://img.shields.io/maven-central/v/co.fs2/fs2-core_2.12.svg)](https://index.scala-lang.org/functional-streams-for-scala/fs2/fs2-core)
 
+### Overview
+
+FS2 is a library for purely functional, effectful, and polymorphic stream processing library in the [Scala programming language]((https://scala-lang.org). Its design goals are compositionality, expressiveness, resource safety, and speed. The name is a modified acronym for **F**unctional **S**treams for **Scala** (FSS, or FS2).
+
+FS2 is available for Scala 2.11, Scala 2.12, and [Scala.js](http://www.scala-js.org/). FS2 is built upon two major functional libraries for Scala, [Cats](https://typelevel.org/cats/), and [Cats-Effect](https://typelevel.org/cats-effect/). Regardless of those dependencies, FS2 core types (streams and pulls) are polymorphic in the effect type (as long as it is compatible with `cats-effect` typeclasses), and thus FS2 can be used with other IO libraries, such as [Monix](https://monix.io/), or [ZIO](https://scalaz.github.io/scalaz-zio/).
+
+Prior to 2016, FS2 was known as `scalaz-stream`, which was based on the [`scalaz`](https://github.com/scalaz/scalaz) library.
```
Comment (mpilquist) [inline README.md]: Prior to the 0.9 release in 2016

File: README.md [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+FS2: Functional Streams for Scala
 =============
 
 [![Build Status](https://travis-ci.org/functional-streams-for-scala/fs2.svg?branch=series/1.0)](http://travis-ci.org/functional-streams-for-scala/fs2)
 [![Gitter Chat](https://badges.gitter.im/functional-streams-for-scala/fs2.svg)](https://gitter.im/functional-streams-for-scala/fs2)
 [![Latest version](https://img.shields.io/maven-central/v/co.fs2/fs2-core_2.12.svg)](https://index.scala-lang.org/functional-streams-for-scala/fs2/fs2-core)
 
+### Overview
+
+FS2 is a library for purely functional, effectful, and polymorphic stream processing library in the [Scala programming language]((https://scala-lang.org). Its design goals are compositionality, expressiveness, resource safety, and speed. The name is a modified acronym for **F**unctional **S**treams for **Scala** (FSS, or FS2).
```
Comment (mpilquist) [inline README.md]: Double opening parents on the link

File: README.md [hunk 3]
```diff
@@ -68,56 +117,12 @@ These features mean that FS2 goes beyond streaming I/O to offer a very general a
 
 ### <a id="docs"></a>Documentation and getting help ###
 
+* Here are the API ScalaDocs for [The core library][core-api], which defines and implements the core types for streams and pulls, as well as the type aliases for pipes and sinks. [The `io` library][io-api], FS2 bindings for NIO-based file I/O and TCP/UDP networking
```
Comment (mpilquist) [inline README.md]: s/Here are the/There are/
s/The core/the core/

File: site/src/main/tut/ecosystem.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+position: 5
+---
+
+### Typelevel Dependencies
+
+As described in the `README` file, FS2 is based mostly on `cats` and `cats-effect`.
+
+* The `cats` library provides definitions for the typeclasses `Functor`, `Applicative`, `Monad`, and `MonadError`.
+* FS2 also uses the data type [`cats.data.Chain`](https://typelevel.org/cats/datatypes/chain.html) from the `cats-core` module. This data structure implements an ordered list-like sequence as a catenable list, which allows for efficient insertion at both ends. This data type was originally developed in `fs2`, where it was called `Catenable`.
+* The `cats-effect` library extends the typeclasses from `cats` with other more specifi typeclasses to describe effectful computations. These type-classes cover resource  bracketing, concurrency, asynchronicity, input-output, timing, and interruption.
```
Comment (mpilquist) [inline site/src/main/tut/ecosystem.md]: s/specifi/specific/

File: site/src/main/tut/ecosystem.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+* FS2 also uses the data type [`cats.data.Chain`](https://typelevel.org/cats/datatypes/chain.html) from the `cats-core` module. This data structure implements an ordered list-like sequence as a catenable list, which allows for efficient insertion at both ends. This data type was originally developed in `fs2`, where it was called `Catenable`.
+* The `cats-effect` library extends the typeclasses from `cats` with other more specifi typeclasses to describe effectful computations. These type-classes cover resource  bracketing, concurrency, asynchronicity, input-output, timing, and interruption. 
+* FS2 also uses from `cats-effect` several data structures used for communicating  and coordinating concurrent processes, such as [semaphores](https://typelevel.org/cats-effect/concurrency/semaphore.html), [deferred values](https://typelevel.org/cats-effect/concurrency/semaphore.html), [atomic references](https://typelevel.org/cats-effect/concurrency/semaphore.html). Some of these types were also were first developed within the internal implementation of FS2, but then moved to `cats-effect`.
```
Comment (mpilquist) [inline site/src/main/tut/ecosystem.md]: Links to deferred and refs are wrong

**Comment by diesalbla** [comment]
@mpilquist Sorry for the late check. Did this ever got deployed? The microsite page does not seem to reflect the changes.

**Comment by mpilquist** [comment]
@diesalbla Apologies for the delay - site was published just now.

**Pull request merged**